### PR TITLE
Implement apiBody and apiQuery tags

### PR DESCRIPTION
**Pull request by SWheeler17**
This is for issue #24 which suggested an apiBody and apiQuery be created instead of using (the general tag) apiParam.

### CODE REVIEW

**Comment by NicolasCARPi** [comment]
Hello,

Thank you for your contribution. It would be great if you could add some tests.

Also, the documentation needs to be modified to reflect the changes and the new available params.

Finally, it looks like you basically took the code from api_param and copy/pasted it to create the two new types. Isn't it possible to reduce this duplication in one way or another?

**Comment by SWheeler17** [comment]
@NicolasCARPi where is the documentation? I was not able to find it in the project, though I may have been looking in the wrong place for it.

**Comment by NicolasCARPi** [comment]
Here you go: https://github.com/apidoc/apidocjs.com

**Comment by SWheeler17** [comment]
@NicolasCARPi I am working on updating the documentation. When I make a PR, do you want me to reference this PR (with a URL or something to that extent)?

**Comment by NicolasCARPi** [comment]
@SWheeler17 This is already much better. I'll merge this after you propose a documentation change. It's important to keep the documentation in sync with the code, and because you are adding functionality, it is also your responsability to modify the documentation to indicate the usage of such params and in which case they are useful.

**Comment by SWheeler17** [comment]
Ok, @NicolasCARPi how do you want me to link this PR to the PR on the documentation PR? I was going to add the link of this PR to that (eventual) PR unless you want me to do something else.

**Comment by NicolasCARPi** [comment]
Whatever works for you!

**Comment by SWheeler17** [comment]
Documentation PR: https://github.com/apidoc/apidocjs.com/pull/34

**Comment by NicolasCARPi** [comment]
Please ping me if this has not been reviewed in a week.

**Comment by SWheeler17** [comment]
@NicolasCARPi this has not been merged and does not appear to have been reviewed either.

File: lib/workers/api_body_title.js [hunk 1]
```diff
@@ -0,0 +1,165 @@
+var semver = require('semver');
```
Comment (tommy87) [inline lib/workers/api_body_title.js:1]: This whole file is mostly a copy of "api_param_title.js". You should base on the existing code (maybe including api_param_title.js) and apply only small changes. 

Having the same code 3 is very bad

File: lib/workers/api_query_title.js [hunk 2]
```diff
@@ -0,0 +1,165 @@
+var semver = require('semver');
```
Comment (tommy87) [inline lib/workers/api_query_title.js:1]: This whole file is mostly a copy of "api_param_title.js". You should base on the existing code (maybe including api_param_title.js) and apply only small changes. 

Having the same code 3 is very bad

**Review by tommy87 — CHANGES_REQUESTED** [review]

**Comment by tommy87** [comment]
Here is a good example https://github.com/apidoc/apidoc-core/blob/master/lib/workers/api_success_title.js

**Comment by SWheeler17** [comment]
@tommy87 I updated the two files you commented on, addressing your comments. Please review it again.

**Review by tommy87 — APPROVED** [review]
Looks much better, but i still need to run tests.

@rottmann @NicolasCARPi 
is there any chance to enable CI / Actions in this project which run the tests automatically?

**Comment by NicolasCARPi** [comment]
@tommy87 Yes sure! Github actions or CircleCI.

**Comment by tommy87** [comment]
@NicolasCARPi ah i thought it must be enabled first, but when it is similar to gitlab i could create a simple test in the next days

**Comment by NicolasCARPi** [comment]
I would favor github actions as they do not require a third party account/access.

There used to be Travis CI IIRC on this repo. But maybe @rottmann pulled the plug.

**Comment by tommy87** [comment]
@SWheeler17 i have create some CI actions on the dev branch. Could you rebase your branch so that the tests are triggered

@NicolasCARPi shouldnt the dokumentation be updated before pulling?

**Comment by SWheeler17** [comment]
@tommy87 I have rebased and the tests seem to have run successfully. The documentation PR is here: https://github.com/apidoc/apidocjs.com/pull/34 (for future reference).

File: test/parser_api_body_test.js [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+ * Test: Parser apiParam
+ */
+
+// node modules
+var should = require('should');
+
+// lib modules
+var parser = require('../lib/parsers/api_body');
+
+describe('Parser: apiParam', function() {
```
Comment (tommy87) [inline test/parser_api_body_test.js:13]: This is the apiBody test and not the apiParam test

**Review by tommy87 — CHANGES_REQUESTED** [review]
Just correct the name and then everythign should be fine

**Comment by SWheeler17** [comment]
Oops, fixed @tommy87, my bad.

**Review by tommy87 — APPROVED** [review]
@NicolasCARPi i think it is ready for pulling

**Comment by SWheeler17** [comment]
What is the likelihood of this PR being pulled within the next few days?

**Comment by NicolasCARPi** [comment]
> What is the likelihood of this PR being pulled within the next few days?

100% :p

**Pull request merged**